pypiserver/__main__.py

```diff
@@ -195,17 +195,33 @@ def main(argv=None):
                 err = sys.exc_info()[1]
                 sys.exit("Invalid port(%r) due to: %s" % (v, err))
         elif k in ("-a", "--authenticate"):
-            c.authenticated = [a.lower()
-                               for a in re.split("[, ]+", v.strip(" ,"))
-                               if a]
-            if c.authenticated == ['.']:
-                c.authenticated = []
+            if '{' in v:
+                import ast
+                v = ast.literal_eval(v)
```

Comment:
@mplanchard I don't follow
> We could then also roll the next block of logic if isinstance(v, dict): into the else for that statement, keeping it all within the if block starting with if '{' in v:, since there's no other case in which v will be a dict anyway.